File: live/src/main/java/com/zky/live/provider/LiveProvider.java
package com.zky.live.provider;

import android.content.Context;
import android.support.v4.app.Fragment;
import com.alibaba.android.arouter.facade.annotation.Route;
import com.zky.basics.common.provider.ILiveProvider;
import com.zky.zky_mine.fragment.LiveMainFragment;

import static com.zky.basics.ArouterPath.ARouterPath.LIVE_MAIN;


@Route(path = LIVE_MAIN, name = "直播")
public class LiveProvider implements ILiveProvider {

    @Override
    public void init(Context context) {

    }



    @Override
    public Fragment getMainLiveFragment() {
        return LiveMainFragment.newInstance();
    }
}

File: live/src/main/java/com/zky/live/provider/LiveProvider.kt
package com.zky.live.provider

import android.content.Context
import androidx.fragment.app.Fragment
import com.alibaba.android.arouter.facade.annotation.Route
import com.zky.basics.ArouterPath.ARouterPath
import com.zky.basics.common.provider.ILiveProvider
import com.zky.zky_mine.fragment.LiveMainFragment.Companion.newInstance

@Route(path = ARouterPath.LIVE_MAIN, name = "直播")
class LiveProvider : ILiveProvider {
    override fun init(context: Context) {}
    override val mainLiveFragment: Fragment
        get() = newInstance()
}